### PR TITLE
added script to check if files contain jsdoc format comments

### LINKED ISSUE #474: Write a script to check if all pull requests meet the requirements for JSDoc
In order to ensure that documentation is seamlessly being created with each addition/modification of a file, write a script to check that the file follows the specifications described in the Getting Started section [here](https://jsdoc.app/). This task essentially involves the creating of a parser (preferred from a maintainability point of view) or use of a regular expression that will check to see if the patterns for generating documentation are adhered to. The script should be written in Python and is written with the goal of it being used with a Github action being triggered upon a pull request.

Parent Issue
- https://github.com/PalisadoesFoundation/talawa-docs/issues/188

### PR DESCRIPTION
**Pull request by akshatgarg12**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #474 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**
Screenshot 2022-03-08 at 12 30 27 PM

**If relevant, did you update the documentation?**

**Summary**
1. added check-jsdocs-comment.py script which checks the file content of all js files in the codebase and looks for comments based on (/** */) this format. if a comment is found, jsdocs pipeline should run to update the documentation, else not.
2. The script sets a env variable RUN_JSDOCS in the workflow file, which can be used with an if statement to check whether to run JSdocs or not.
3. Currently JSdocs pipeline is not created as it is a part of separate child issue, but the logic has been written and can be implemented over it.

**Does this PR introduce a breaking change?**

No
**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by akshatgarg12** [comment]
**Checking workflow files / Checking for unauthorized file changes (pull_request)** check is failing because [this .github/workflow file ](https://github.com/PalisadoesFoundation/talawa-api/blob/develop/.github/workflows/authorized-changes-detection.yml) does not allow any changes in **/.github** folder in a PR.

**Comment by palisadoes** [comment]
With each PR we check to ensure that we have tests for as much of the code base as possible, this is the code coverage test. This PR would not not be applicable as your code isn't JavaScript based. This is OK.

Yes, you are right, we have checks to make sure the `.github` directory isn't updated as this is an area where most of our automation resides. This PR hopes to improve our automation, so this is acceptable.

**Review by palisadoes — DISMISSED** [review]

**Review by DMills27 — COMMENTED** [review]
Can you add a more detailed comment about what the Regex is doing? For anyone reading this in the future they might be uncertain as to how exhaustive this pattern match is.

**Comment by akshatgarg12** [comment]
> Can you add a more detailed comment about what the Regex is doing? For anyone reading this in the future they might be uncertain as to how exhaustive this pattern match is.

sure

File: .github/workflows/check-jsdocs-comment.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            if matches:
+                comments_for_jsdoc_exists = True
+                break
+    # setting github env variable to decide to run jsdoc pipeline
+    try:
+        with open(env_file, "a") as myfile:
+            if comments_for_jsdoc_exists == True:
+                myfile.write("RUN_JSDOCS=True")
+            else:
+                myfile.write("RUN_JSDOCS=False")
```
Comment (palisadoes) [inline .github/workflows/check-jsdocs-comment.py:50]: Please standardize on single quotes for all strings.

File: .github/workflows/check-jsdocs-comment.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
```
Comment (palisadoes) [inline .github/workflows/check-jsdocs-comment.py:1]: Please add Docstrings to all your functions and also at the top of your script like [countline.py](https://github.com/PalisadoesFoundation/talawa-api/blob/develop/.github/workflows/countline.py)

This script was created to ensure that our code base has correct documentation. The script should also have good internal documentation too.

Make sure the script is pydocstring compliant

File: .github/workflows/check-jsdocs-comment.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+def main():
+    # get current working dir
+    directory = os.getcwd()
+    # this is to setup workflow env variable
+    env_file = os.getenv('GITHUB_ENV')
+    # The regex checks if the content of files contain a comment of form => /** {charachters | newline} */ ,
+    # file may also contain any number of newlines or charachters before or after the comment. 
+    # To analyse the regex visit https://regex101.com/ and paste the regex
+    pattern = re.compile("^(.|\n)*(\/\*\*(.|\n)*\*\/)+(.|\n)*$");
+    # list of dir to look in repo for files
```
Comment (palisadoes) [inline .github/workflows/check-jsdocs-comment.py:16]: Make sure the script is pylint, pycodestyle and flake8 compliant.

**Review by palisadoes — CHANGES_REQUESTED** [review]

**Review by palisadoes — APPROVED** [review]

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -52,6 +52,20 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  JSDocs:
+    name : "JSDocs comments and pipeline"
+    runs-on : ubuntu-latest
```
Comment (palisadoes) [inline .github/workflows/pull-request.yml:57]: Make this rely on `Code-Formatter` like `Code-Coverage` below. The correct functioning of this Action will probably rely on correct formatting. So we will need to make sure the ordering is maintained.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments in `pull-request.yml`

**Review by palisadoes — APPROVED** [review]

**Pull request merged**